modules/client/src/main/java/org/apache/ignite/internal/jdbc/JdbcStatement.java

```diff
@@ -734,16 +733,6 @@ public final void timeout(int timeout) throws SQLException {
     }
 
     private static SQLException toSqlException(CompletionException e) {
-        if (e.getCause() instanceof IgniteException) {
-            IgniteException cause = (IgniteException) e.getCause();
-            String message = cause.getMessage();
-
-            if (message != null) {
-                if (message.contains("Failed to parse query")) {
-                    return new SQLException("Sql query execution failed.", SqlStateCode.PARSING_EXCEPTION, e);
-                }
-            }
-        }
-        return new SQLException("Internal server error.", SqlStateCode.INTERNAL_ERROR, e);
+        return new SQLException(e.getCause());
```

Comment:
It does not seem correct. You should not lose original `CompletionException` because it will lead to the fact that its stack frame will be lost as well. Let's change this line to `return new SQLException(e));`